src/helpers/message.ts

```diff
@@ -27,50 +28,56 @@ function _getTokenFmt(amount: BigNumber, token: TokenDto) {
 }
 
 // TODO: probably there is (or there should be) a function shared on dex-js for this
-export function calculatePrice(order: OrderDto): BigNumber {
+export function calculatePrice(
+  order: Pick<OrderDto, 'buyToken' | 'sellToken' | 'priceNumerator' | 'priceDenominator'>,
+): BigNumber {
   const { buyToken, sellToken, priceNumerator, priceDenominator } = order
+  const buyTokenDecimals = buyToken.decimals
+  const sellTokenDecimals = sellToken.decimals
 
-  if (buyToken.decimals >= sellToken.decimals) {
-    const precisionFactor = 10 ** (buyToken.decimals - sellToken.decimals)
-    return priceNumerator.dividedBy(priceDenominator.multipliedBy(precisionFactor))
+  let price
+  if (buyTokenDecimals >= sellToken.decimals) {
+    const precisionFactor = 10 ** (buyTokenDecimals - sellTokenDecimals)
+    price = priceNumerator.dividedBy(priceDenominator.multipliedBy(precisionFactor))
   } else {
-    const precisionFactor = 10 ** (sellToken.decimals - buyToken.decimals)
-    return priceNumerator.multipliedBy(precisionFactor).dividedBy(priceDenominator)
+    const precisionFactor = 10 ** (sellTokenDecimals - buyTokenDecimals)
+    price = priceNumerator.multipliedBy(precisionFactor).dividedBy(priceDenominator)
   }
+
+  return price.decimalPlaces(buyTokenDecimals, BigNumber.ROUND_FLOOR)
```

Comment:
Is it not a problem to adjust the price only based on `buyTokenDecimals`?

Reply:
At the beginning I was using both, taking the maximum of buyTokenDecimals, sellTokenDecimals.

Then I realised that the price is used to calculate the receiving amount, that are expressed in buy tokens. So I've used the precision of the buy token and round down (slightly worse price)

Reply:
Now that I think about this again, I'm not 100% sure of the decimals we can have in the price.

I'm starting to think that my approach could fail with sell amounts with decimals.

Let me formulate it:
* We are selling `X` tokens for `Y` tokens at a price `P` => `P x X = Y`
* `precision(x)` is the precision of the token `x`
* `decimals(x)` is the number of decimals in `x`

We know:
* `decimals(x) < precision(x), for X and Y`

Ideally we want to find a P:
* That `decimals(P x X) <= precision(Y)`

Reply:
Worst case is:
`decimals(P x X) = decimals(P) +  decimals(X)`

Therefore, there's not such `P` 😔

...unless we do this price calculation in two steps:

Let's see an example:
```
* decimals(A) = 2
* decimals(A) = 1
* amount A = 1.23

P = 9.87654321
P x A = 1.23 * 9.837398374 = 12.1481481483
B = 12.1

Then we adjust P to satisfy:
1.23 * P = 12.1
P = 12.1 / 1.23 = 9.837398374
```

Reply:
I'll make a separate PR for this